doc/tutorials/contributing-windows.md

```diff
@@ -0,0 +1,62 @@
+# Introduction
+
+In this tutorial, you will learn how to set up a development environment for Elektra using Windows Subsystem for Linux (WSL) and Visual Studio 2019.
+For further information, consider the
+[Linux with Visual Studio C++ documentation](https://docs.microsoft.com/en-us/cpp/linux/?view=msvc-160).
+
+Visual Studio 2019 (VS2019) Community is free but requires a registration after a couple of days.
+[Download it](https://visualstudio.microsoft.com/downloads/)
+and install the Linux development with C++ workload.
+
+Elektra development mainly happens on GNU/Linux distributions.
+Therefore, it makes sense to use WSL if you have Windows 10 installed on your PC.
+For the beginning, it does not matter if you choose WSL 1 or WSL 2.
```

Comment:
I would strongly recommend WSL 2 over WSL 1 for Elektra development. The IO performance of WSL 1 is pretty bad, this could make for pretty bad/slow experience. Also since WSL 2 is a full Linux kernel, even the more complex plugins should work out of the box (AFAIK e.g. dbus has some problems under WSL 1)

Reply:
> Also since WSL 2 is a full Linux kernel, even the more complex plugins should work out of the box

I have no idea about that. I just hit "Build all" with the defaults for the moment.

According to [a blog post from Microsoft ](https://devblogs.microsoft.com/cppblog/c-with-visual-studio-and-wsl2/) performance **across both filesystems** should be faster with WSL1. We could check how good that SSH option from that blog post works.

Reply:
The blog post you linked talks about accessing the NTFS C: drive via the Linux mount `/mnt/c`. This is faster in WSL1. I assumed that people would close Elektra into e.g. `~/elektra`. Then the files are part of the native Linux file system. In that case file IO should me much faster in WSL2, [as outlined here](https://docs.microsoft.com/en-us/windows/wsl/compare-versions#increased-file-io-performance).

But if we don't want to recommend anything, we could just link to [the official comparison](https://docs.microsoft.com/en-us/windows/wsl/compare-versions). Then people can decide for themselves.

Reply:
>  I assumed that people would close Elektra into e.g. ~/elektra.

I assumed the contrary 😄  because
1. when you open VS2019 then the Windows Explorer opens and 
2.   The [docu](https://docs.microsoft.com/en-us/cpp/linux/cmake-linux-configure?view=msvc-160) at some point said
> When you target WSL, Visual Studio doesn't need to copy source files and maintain two synchronous copies of your build tree because the compiler on Linux has direct access to your source files in the mounted Windows file system.

Nevertheless, you're right about I/O and cloning into Linux filesystem certainly makes sense, too.  What I wrote down was just the first way that I got everything to compile.